rapids-cmake/export/detail/component.cmake

```diff
@@ -0,0 +1,80 @@
+#=============================================================================
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_export_component
+-----------------------
+
+.. versionadded:: v23.04.00
+
+.. code-block:: cmake
+
+Generate the necessary -Config.cmake modules needed for an optional component
+of a rapids export package.
+
+  rapids_export_component( (BUILD|INSTALL) <project_name> <component_name> <export_set> <unique_name> <namespace>)
+
+The :cmake:command:`rapids_export_component` function generates
+the `<proj>-<unique_name>-targets.cmake` and `<proj>-<unique_name>-dependencies.cmake`
+files necessary for :cmake:command:`rapids_export` to support optional
+components.
+
+.. note::
+  It is an anti-pattern to use this function directly. The vast majority of projects
```

Comment:
Are there any real use cases of invoking this function directly? Since it's a detail function I default to assuming it should never need to be used. If there are niche use cases maybe it should be moved out of detail and instead documented as advanced a la `rapids_export_package` or `rapids_export_find_package_file`.

Reply:
There are no real uses cases of invoking this function directly. I am happy to drop that comment

Reply:
Yes please do.